### PR TITLE
prevent potential XSS from searchbar results

### PR DESCRIPTION
**Pull request by diegocepedaw**
Tested against exploit reported in issue #341

### CODE REVIEW

File: src/oncall/ui/static/js/oncall.js [hunk 1]
```diff
@@ -3484,6 +3484,19 @@ var oncall = {
         });
       });
     }
+  },
+  sanitize: function (string) {
```
Comment (colinxy) [inline src/oncall/ui/static/js/oncall.js:3488]: I can see this sanitize is useful when putting user input into html, but we are forming an url from user input. Have you tested the report's user input with your changes yet? What is the resulting url? Would appreciate if you can post the test results as well.

I am not expert on XSS, but I think url/percent encoding may be more appropriate.

**Review by colinxy — CHANGES_REQUESTED** [review]

File: src/oncall/ui/static/js/oncall.js (hunk 1 above)
Comment (jrgp) [inline src/oncall/ui/static/js/oncall.js:3488]: Why not use the Handlebars built-in escape function (afaik Oncall still uses handlebars)? Or just use Handlebars formatting for these strings?

**Review by jrgp — CHANGES_REQUESTED** [review]

File: src/oncall/ui/static/js/oncall.js (hunk 1 above)
Comment (diegocepedaw) [reply src/oncall/ui/static/js/oncall.js:3488]: Good suggestion, Handlebars.escapeExpression(string) does the exact same thing as this sanitize function so I might as well use that. https://handlebarsjs.com/api-reference/utilities.html#handlebars-escapeexpression-string

File: src/oncall/ui/static/js/oncall.js [hunk 2]
```diff
@@ -579,11 +579,11 @@ var oncall = {
               },
               footer: function(resp){
                 if (teamsCt > typeaheadLimit) {
-                  return '<div class="tt-see-all"><a href="/query/' + resp.query + '/teams" data-navigo> See all ' + teamsCt + ' results for teams »</a></div>';
+                  return '<div class="tt-see-all"><a href="/query/' + Handlebars.escapeExpression(resp.query) + '/teams" data-navigo> See all ' + teamsCt + ' results for teams »</a></div>';
```
Comment (jrgp) [inline src/oncall/ui/static/js/oncall.js:582]: Like Colin mentioned, for places where resp.query is being interpolated into a URL, you should probably wrap it in encodeURIComponent() as well, to ensure URL-specific characters are escaped into percent notation, generating a working URL. As in:

```
'<div class="tt-see-all"><a href="/query/' + Handlebars.escapeExpression(encodeURIComponent(resp.query)) + ' etc etc '
```

**Review by jrgp — CHANGES_REQUESTED** [review]

File: src/oncall/ui/static/js/oncall.js (hunk 2 above)
Comment (diegocepedaw) [reply src/oncall/ui/static/js/oncall.js]: added

File: src/oncall/ui/static/js/oncall.js (hunk 1 above)
Comment (diegocepedaw) [reply src/oncall/ui/static/js/oncall.js]: changed to use inbuild handlebars functions. Tested in existing names and services as well as names and services that include special characters.

**Review by colinxy — APPROVED** [review]

**Pull request merged**